spring-shell-samples/src/main/java/org/springframework/shell/samples/standard/ComponentCommands.java

```diff
@@ -93,7 +93,7 @@ public String singleSelector() {
 	public String multiSelector() {
 		List<SelectorItem<String>> items = new ArrayList<>();
 		items.add(SelectorItem.of("key1", "value1"));
-		items.add(SelectorItem.of("key2", "value2", false));
+		items.add(SelectorItem.of("key2", "value2", false, true));
```

Comment:
This would also need a change in https://github.com/spring-projects/spring-shell/blob/643b189fb8d63f9a7aeee361f12a30301864e9a2/spring-shell-core/src/main/resources/org/springframework/shell/component/multi-item-selector-default.stg#L21 as it was assumed that if item is disabled it would never show selected/disabled style. I'd also change below key3 to selected to demonstrate it with item which can be changed.

Reply:
Do we need new styling here? So as far as I can see, if it is disabled we currently would only be using the "style-item-disabled", even it was selected right?

Do you have an idea how this could be styled? Currently there are the following for stages:
- enabled & selected
- enabled & not selected
- disabled 

instead of the disabled we would have:
- disabled & selected
- disabled & not selected

The disabled & not selected could be the same styling (as it currently is), but is is simply enough to add the checkbox if it is selected?

```
<if(item.enabled)>
  <if(item.selected)>
    <({<figures.checkboxOn> }); format=selected_style(item.selected)> <item.name>
  <else>
    <({<figures.checkboxOff> }); format=selected_style(item.selected)> <item.name>
  <endif>
<else>	
  <if(item.selected)>
     <({<figures.checkboxOn> }); format="style-item-disabled"> <item.name; format="style-item-disabled">
  <else>
    <({<figures.checkboxOff> }); format="style-item-disabled"> <item.name; format="style-item-disabled">
  <endif>  
<endif>
```


Could you also give me a hint how I can test the format changes? Building the application is quite cumbersome and takes some time just to test if the styling is appropriate.

Reply:
Yes it should be as simple as what you showed. Testing these are difficult as it's not easy when ansi escape characters are involved and `MultiItemSelectorTests` takes you only that far. I'd just try it out manually until we get something better for tests(I've been working on a node bases e2e tests which is sometimes only way to test what user is seeing).

Reply:
![image](https://user-images.githubusercontent.com/7412602/179826751-7e9862f2-c500-48f2-8117-6cbb18cef7c5.png)

Actually I think that looks quite nice.